### PR TITLE
add no7singing

### PR DESCRIPTION
**Pull request by frankxu2004**

### CODE REVIEW

File: egs/kiritan/svs1/conf/tuning/train_naive_rnn.yaml [hunk 1]
```diff
@@ -76,3 +76,6 @@ best_model_criterion:
 -   - train
     - loss
     - min
+
+gpu_id: 0
```
Comment (ftshijt) [inline egs/kiritan/svs1/conf/tuning/train_naive_rnn.yaml:80]: As we decided to normalize the scripts, local information like `gpu_id` or `vocoder_checkpoint` would not be present in the config

File: egs/kiritan/svs1/conf/tuning/train_naive_rnn.yaml [hunk 2]
```diff
@@ -76,3 +76,6 @@ best_model_criterion:
...
+vocoder_checkpoint: /home/fangzhex/music/train_nodev_kiritan_hifigan.v1/checkpoint-300000steps.pkl
```
Comment (ftshijt) [inline egs/kiritan/svs1/conf/tuning/train_naive_rnn.yaml:81]:
```suggestion
```

File: egs/kiritan/svs1/run.sh [hunk 3]
```diff
@@ -28,7 +28,10 @@ valid_set=dev
 test_sets="dev eval"
 
 # training and inference configuration
-train_config=conf/train.yaml
+train_config=conf/tuning/train_naive_rnn.yaml
+# train_config=conf/tuning/train_xiaoice.yaml
+# train_config=conf/tuning/train_xiaoice_noDP.yaml
+# train_config=conf/train.yaml
```
Comment (ftshijt) [inline egs/kiritan/svs1/run.sh:34]:
```suggestion
train_config=conf/train.yaml
```

Same reason

File: egs/kiritan/svs1/run.sh [hunk 4]
```diff
@@ -55,5 +58,7 @@ cleaner=none
     --test_sets "${test_sets}" \
     --score_feats_extract "${score_feats_extract}" \
     --srctexts "data/${train_set}/text" \
+    --svs_exp ${expdir} \
+    --vocoder_file "/home/fangzhex/music/train_nodev_kiritan_hifigan.v1/checkpoint-300000steps.pkl" \
```
Comment (ftshijt) [inline egs/kiritan/svs1/run.sh:62]:
```suggestion
```

File: egs/no7singing/svs1/conf/train.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+tuning/train_glu_transformer.yaml
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/train.yaml:1]: is there a specific reason that you use glu_transformer for default training config? If not, recommend to be naive_rnn

File: egs/no7singing/svs1/conf/tuning/train_bytesing.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_bytesing.yaml:1]: bytesing is deprecated in our recipes, please remove it

File: egs/no7singing/svs1/conf/tuning/train_glu_transformer.yaml [hunk 7]
```diff
@@ -0,0 +1,84 @@
+
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_glu_transformer.yaml:1]: We have changed the style for config, please refer to the other recipes for glu-transformer config

File: egs/no7singing/svs1/conf/tuning/train_mlp_singer.yaml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+best_model_criterion:
+-   - valid
+    - loss
+    - min
+-   - train
+    - loss
+    - min
+
+gpu_id: 1
+vocoder_checkpoint: /data1/gs/peter/ParallelWaveGAN/egs/singing/voc1/exp/train_hifigan.v1_train_nodev_clean_libritts_hifigan-2.v1/checkpoint-50000steps.pkl
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_mlp_singer.yaml:69]:
```suggestion
```

File: egs/no7singing/svs1/conf/tuning/train_mlp_singer.yaml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+##########################################################
+num_iters_per_epoch: 200    # number of iterations per epoch
+max_epoch: 500              # number of epochs
+grad_clip: 1.0              # gradient clipping norm
+grad_noise: false           # whether to use gradient noise injection
+accum_grad: 1               # gradient accumulation
+
+# batch_bins: 1000000       # batch bins (for feats_type=fbank)
+# batch_bins: 10000000         # batch bins (for feats_type=raw, *= n_shift / n_mels)
+# batch_type: numel           # how to make batch
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_mlp_singer.yaml:47]:
```suggestion
```

File: egs/no7singing/svs1/conf/tuning/train_naive_rnn.yaml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+##########################################################
+# num_iters_per_epoch: 200    # number of iterations per epoch
+max_epoch: 500              # number of epochs
+grad_clip: 1.0              # gradient clipping norm
+grad_noise: false           # whether to use gradient noise injection
+accum_grad: 1               # gradient accumulation
+
+# batch_bins: 1000000       # batch bins (for feats_type=fbank)
+# batch_bins: 10000000         # batch bins (for feats_type=raw, *= n_shift / n_mels)
+# batch_type: numel           # how to make batch
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_naive_rnn.yaml:56]:
```suggestion
```

File: egs/no7singing/svs1/conf/tuning/train_naive_rnn.yaml [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+-   - valid
+    - loss
+    - min
+-   - train
+    - loss
+    - min
+
+gpu_id: 0
+vocoder_checkpoint: /home/fangzhex/music/train_nodev_kiritan_hifigan.v1/checkpoint-300000steps.pkl
+# vocoder_checkpoint: /data5/gs/vocoder_peter/hifigan-vocoder/exp/train_hifigan.v1_train_nodev_clean_libritts_hifigan-2.v1/checkpoint-50000steps.pkl
```
Comment (ftshijt) [inline egs/no7singing/svs1/conf/tuning/train_naive_rnn.yaml:86]:
```suggestion
```

File: egs/no7singing/svs1/local/dataset_split.py [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+# midfiles = list(find_files_by_extensions(data_root_url, exts=[".mid"]))
+# for path in midfiles:
+#     if path[-6:] == "13.mid":
+#         path13 = path
+#     if path[-6:] == "14.mid":
+#         path14 = path
+# path_temp = path14[-6:] + "00.mid"
+# shutil.move(path13, path_temp)
+# shutil.move(path14, path13)
+# shutil.move(path_temp, path14)
```
Comment (ftshijt) [inline egs/no7singing/svs1/local/dataset_split.py:129]:
```suggestion
```

File: egs/no7singing/svs1/local/dataset_split.py [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Prepare segments from HTS-style alignment files",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("db_root", type=str, help="source dataset midi folder path")
+    parser.add_argument("save_dir", type=str, help="path of dataset split for save")
+    parser.add_argument("dev", type=float, help="absolute size of dev dataset")
+    parser.add_argument("eval1", type=float, help="absolute size of test dataset")
```
Comment (ftshijt) [inline egs/no7singing/svs1/local/dataset_split.py:35]: I don't think these two arguments are necessary given that we have defined the DEV_LIST and TEST_LIST

File: egs/no7singing/svs1/run.sh [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+train_set=tr_no_dev
+valid_set=dev
+test_sets="dev eval"
+
+# training and inference configuration
+train_config=conf/tuning/train_naive_rnn.yaml
+# train_config=conf/tuning/train_xiaoice.yaml
+# train_config=conf/tuning/train_xiaoice_noDP.yaml
+# train_config=conf/train.yaml
```
Comment (ftshijt) [inline egs/no7singing/svs1/run.sh:36]:
```suggestion
train_config=conf/train.yaml
```

File: egs/no7singing/svs1/run.sh [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+fs=24000
+fmin=80
+fmax=7600
+n_fft=2048
+n_shift=300
+win_length=1200
+
+
+score_feats_extract=frame_score_feats   # frame_score_feats | syllable_score_feats
+expdir=exp/rnn_test
```
Comment (ftshijt) [inline egs/no7singing/svs1/run.sh:18]:
```suggestion
```

File: egs/no7singing/svs1/run.sh [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+    --cleaner ${cleaner} \
+    --train_config "${train_config}" \
+    --inference_config "${inference_config}" \
+    --train_set "${train_set}" \
+    --valid_set "${valid_set}" \
+    --test_sets "${test_sets}" \
+    --score_feats_extract "${score_feats_extract}" \
+    --srctexts "data/${train_set}/text" \
+    --svs_exp ${expdir} \
+    --vocoder_file "/home/fangzhex/music/ParallelWaveGAN/egs/no7singing/voc1/exp/train_nodev_no7singing_hifigan.v1/checkpoint-240000steps.pkl" \
```
Comment (ftshijt) [inline egs/no7singing/svs1/run.sh:65]:
```suggestion
```

File: tools/activate_python.sh [hunk 17]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
```
Comment (ftshijt) [inline tools/activate_python.sh:1]: Please do not change this file (it is left for users to install)

**Review by ftshijt — COMMENTED** [review]
Many thanks:
some minor comments as follows

**Comment by frankxu2004** [comment]
@ftshijt fixed the issues, please review

**Comment by ftshijt** [comment]
Many thanks!

**Pull request merged**